packages/jest/src/lib/error.ts

```diff
@@ -154,7 +154,7 @@ export function formatError(
       if (
         // @ts-ignore
         !error.snippet &&
-        (!file || fs.realpathSync(file) !== location.file)
+        (!file || fs.realpathSync(`${rootDir}/${file}`) !== location.file)
```

Comment:
@miguelangaranocurrents great for finding the root cause here.

Additionally, because realpathSync throws an error when file not found, we should have a try catch around this if. So it doesn't break the test like we saw.